src/utils/__tests__/enableWhen/equal.test.ts

```diff
@@ -0,0 +1,152 @@
+                {
+                    linkId: 'q2',
+                    answer: [],
+                },
+                {
+                    linkId: CONTROL_ITEM_LINK_ID,
+                    answer: [],
+                },
+            ],
+        }),
+    },
+    {
+        ...generateQAndQRData({
+            type: 'integer',
+            enableWhen: [
+                {
+                    question: 'q1',
+                    operator: '=',
+                    answer: { string: 'test' },
+                },
+            ],
+            qrItem: [
+                {
+                    linkId: 'q1',
+                    answer: [{ value: { string: 'test2' } }],
+                },
+                {
+                    linkId: 'q2',
+                    answer: [],
+                },
+            ],
+        }),
+    },
+    {
+        ...generateQAndQRData({
+            type: 'integer',
+            enableWhen: [
+                {
+                    question: 'q1',
+                    operator: '=',
+                    answer: { string: 'test1' },
+                },
+                {
+                    question: 'q2',
+                    operator: '=',
+                    answer: { string: 'test2' },
+                },
+            ],
+            qrItem: [
+                {
+                    linkId: 'q1',
+                    answer: [{ value: { string: 'test1' } }],
+                },
+                {
+                    linkId: 'q2',
+                    answer: [{ value: { string: 'test2' } }],
+                },
+                {
+                    linkId: CONTROL_ITEM_LINK_ID,
+                    answer: [],
+                },
+            ],
+        }),
+    },
+    {
+        ...generateQAndQRData({
+            type: 'integer',
+            enableWhen: [
+                {
+                    question: 'q1',
+                    operator: '=',
+                    answer: { string: 'test1' },
+                },
+                {
+                    question: 'q2',
+                    operator: '=',
+                    answer: { string: 'test2' },
+                },
+            ],
+            qrItem: [
+                {
+                    linkId: 'q1',
+                    answer: [{ value: { string: 'asd' } }],
+                },
+                {
+                    linkId: 'q2',
+                    answer: [{ value: { string: 'test2' } }],
+                },
+            ],
+        }),
+    },
+    {
+        ...generateQAndQRData({
+            type: 'integer',
+            enableBehavior: 'any',
+            enableWhen: [
+                {
+                    question: 'q1',
+                    operator: '=',
+                    answer: { Coding: { code: 'test1', display: 'test1' } },
+                },
+                {
+                    question: 'q2',
+                    operator: '=',
+                    answer: { Coding: { code: 'test2', display: 'test2' } },
```

Comment:
add a test that checks equality of Codings with different displays but the same code